presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -306,6 +307,18 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
+                if (node.getSortItems().isPresent()) {
+                    for (SortItem sortItem : node.getSortItems().get()) {
+                        Expression sortKey = sortItem.getSortKey();
```

Comment:
Shouldn't you also verify that `sortKey` is column reference. Can you have query similar to:
```
SELECT function(something ORDER BY (CAST(ROW(1) AS ROW(someField BIGINT))).someField) FROM ...
SELECT function(something ORDER BY a.someField) FROM (VALUES CAST(ROW(1) AS ROW(someField BIGINT))) t(a)
```
(you get the idea)?